napari/layers/base/base.py

```diff
@@ -999,6 +1009,131 @@ def _data_to_world(self) -> Affine:
         """
         return self._transforms[1:3].simplified
 
+    def vector_world_to_data(self, vector) -> tuple:
+        """Convert a vector defining an orientation from world coordinates to data coordinates.
+        For example, this would be used to convert the view ray.
+
+        Parameters
+        ----------
+        vector : tuple, list, 1D array
+            A vector in world coordinates.
+
+        Returns
+        -------
+        tuple
+            Vector in data coordinates.
+        """
+        vector = np.asarray(vector)
+
+        # create transform for view direction (world -> layer data)
+        shear_matrix = expand_upper_triangular(self.shear)
+        rot_matrix = self.rotate
+        scale_matrix = np.diag(self.scale)
+        world_to_layer = np.linalg.inv(
+            shear_matrix @ rot_matrix
+        ).T @ np.linalg.inv(scale_matrix)
+
+        return tuple(vector @ world_to_layer)
+
+    @property
+    def _display_bounding_box(self):
+        """An axis aligned (self._ndisplay, 2) bounding box around the data"""
+        return self._extent_data[:, self._dims_displayed].T
+
+    def _data_face_intersection(
+        self,
+        vertices: np.ndarray,
+        click_pos_data: np.ndarray,
+        view_dir_data: np.ndarray,
+    ):
```

Comment:
Does this need to be method on layer, looks like it could be its own function?

Reply:
I agree this should be its own function in our geometry utils!

Reply:
I moved it to our `utils.geometry` module